client-keycloak/src/main/java/org/apache/atlas/keycloak/client/service/AtlasKeycloakAuthService.java

```diff
@@ -0,0 +1,109 @@
+package org.apache.atlas.keycloak.client.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.*;
+import okhttp3.logging.HttpLoggingInterceptor;
+import org.apache.atlas.exception.AtlasBaseException;
+import org.apache.atlas.keycloak.client.RetrofitKeycloakClient;
+import org.apache.atlas.keycloak.client.config.KeycloakConfig;
+import org.jetbrains.annotations.NotNull;
+import org.keycloak.representations.AccessTokenResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+
+import java.time.OffsetDateTime;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.atlas.AtlasErrorCode.BAD_REQUEST;
+
+public final class AtlasKeycloakAuthService {
+
+    public final static Logger LOG = LoggerFactory.getLogger(AtlasKeycloakAuthService.class);
+
+    private final static String GRANT_TYPE = "grant_type";
+    private static final String CLIENT_ID = "client_id";
+    private static final String CLIENT_SECRET = "client_secret";
+    private static final long DEFAULT_MIN_VALIDITY_MINS = 30;
+    private static final int EXPIRY_OFFSET_SEC = 300;
+
+    private final RetrofitKeycloakClient retrofit;
+    private final KeycloakConfig keycloakConfig;
+    private AccessTokenResponse currentAccessToken;
+    private long expirationTime = -1;
+
+    public AtlasKeycloakAuthService(KeycloakConfig keycloakConfig) {
+        this.keycloakConfig = keycloakConfig;
+        this.retrofit = new Retrofit.Builder().client(getOkHttpClient())
+                .baseUrl(this.keycloakConfig.getAuthServerUrl())
+                .addConverterFactory(JacksonConverterFactory.create(new ObjectMapper())).build()
+                .create(RetrofitKeycloakClient.class);
+    }
+
+    @NotNull
+    private OkHttpClient getOkHttpClient() {
+        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
+        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
+        return new OkHttpClient.Builder()
+                .addInterceptor(interceptor)
+                .addInterceptor(errorHandlingInterceptor)
+                .connectTimeout(60, TimeUnit.SECONDS)
```

Comment:
Store this timeout duration in constants & refer in AbstractKeycloakClient as well